src/main/java/sg/ncl/MainController.java

```diff
@@ -1319,45 +1323,72 @@ public String teams(Model model, HttpSession session) {
-//            Team2 team2 = extractTeamInfo(teamInfoObject.toString());
-//            teamManager2.addTeamToPublicTeamMap(team2);
-//        }
+        // check if inner image map is empty, have to do it via this manner
+        // returns true if the team contains an image list
+        isInnerImageMapPresent = imageMap.values().stream().filter(perTeamImageMap -> !perTeamImageMap.isEmpty()).findFirst().isPresent();
 
         model.addAttribute("userEmail", userEmail);
         model.addAttribute("teamMap2", teamManager2.getTeamMap());
-//        model.addAttribute("publicTeamMap2", teamManager2.getPublicTeamMap());
         model.addAttribute("userJoinRequestMap", teamManager2.getUserJoinRequestMap());
-        model.addAttribute("savedImageList", savedImageList);
+        model.addAttribute("isInnerImageMapPresent", isInnerImageMapPresent);
+        model.addAttribute("imageMap", imageMap);
         return "teams";
     }
 
+    /**
+     * Exectues the service-image and returns a Map containing the list of images in two partitions.
+     * One partition contains the list of already created images.
+     * The other partition contains the list of currently saving in progress images.
+     * @param teamId The ncl team id to retrieve the list of images from.
+     * @return Returns a Map containing the list of images in two partitions.
+     */
+    private Map<String, List<Image>> invokeAndGetImageList(String teamId) {
+        Map<String, List<Image>> resultMap = new HashMap<>();
+        List<Image> createdImageList = new ArrayList<>();
+        List<Image> inProgressImageList = new ArrayList<>();
+
+        HttpEntity<String> imageRequest = createHttpEntityHeaderOnly();
+        ResponseEntity imageResponse = restTemplate.exchange(properties.getTeamSavedImages(teamId), HttpMethod.GET, imageRequest, String.class);
+        String imageResponseBody = imageResponse.getBody().toString();
+
+        String osImageList = new JSONObject(imageResponseBody).getString(teamId);
+        JSONObject osImageObject = new JSONObject(osImageList);
+
+        log.info("osImageList: {}", osImageList);
+        log.info("osImageObject: {}", osImageObject);
+
+        if (osImageObject == JSONObject.NULL || osImageObject.length() == 0) {
+            log.info("Image list for Team: {} is empty.", teamId);
+            return resultMap;
+        }
+
+        for (int k = 0; k < osImageObject.names().length(); k++) {
+            String imageName = osImageObject.names().getString(k);
+            String imageStatus = osImageObject.getString(imageName);
+
+            log.info("image name: {} image status: {}", imageName, imageStatus);
+
+            Image image = new Image();
+            image.setImageName(imageName);
+            image.setDescription("-");
+            image.setTeamId(teamId);
+
+            if (imageStatus.equals("created")) {
+                createdImageList.add(image);
+            } else if (imageStatus.equals("notfound")) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move the "notfound" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)